components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIKeyValidator.java

```diff
@@ -353,8 +357,17 @@ public boolean isAPIResourceValidationEnabled() {
     public String getResourceAuthenticationScheme(MessageContext synCtx) throws APISecurityException {
 
         VerbInfoDTO verb = null;
+        TracingSpan span = null;
         try {
+            if (Util.tracingEnabled()) {
```

Comment:
The error cases are not handled with spans in this try block.

Reply:
Fixed